src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala

```diff
@@ -676,6 +681,35 @@ trait SyncControllerBuilder {
 
 }
 
+trait PortForwardingBuilder {
+  self: DiscoveryConfigBuilder =>
+
+  import monix.execution.Scheduler.Implicits.global
+
+  private val portForwardingRelease = new AtomicReference(Option.empty[Future[Task[Unit]]])
+
+  def startPortForwarding(): Unit =
+    portForwardingRelease.compareAndSet(
+      None,
+      Some {
+        PortForwarder
+          .openPorts(
+            Seq(Config.Network.Server.port),
+            Seq(discoveryConfig.port).filter(_ => discoveryConfig.discoveryEnabled)
+          )
+          .whenA(Config.Network.automaticPortForwarding)
+          .allocated
+          .runToFuture
+          .map(_._2)
+      }
+    )
+
+  def stopPortForwarding(): Future[Unit] =
+    portForwardingRelease.get().fold(Future.unit) {
+      _.flatMap(_.runToFuture)
+    }
```

Comment:
This should also set the reference back to `None`, so it can be started again. See [this](https://github.com/input-output-hk/mantis/pull/929#discussion_r582019832) comment for a better way of doing it than the original proposal.